File: code_challenge_solution/java/74 Geekcode generator.java

import java.util.Scanner;

public class GeekcodeGenerator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter your name: ");
        String name = scanner.nextLine();
        
        System.out.print("Enter your favorite programming language: ");
        String language = scanner.nextLine();
        
        System.out.print("Enter your favorite operating system: ");
        String os = scanner.nextLine();
        
        System.out.print("Enter your level of geekiness (1-10): ");
        int geekiness = scanner.nextInt();
        
        String geekcode = generateGeekcode(name, language, os, geekiness);
        System.out.println("Your Geekcode signature: " + geekcode);
    }
    
    public static String generateGeekcode(String name, String language, String os, int geekiness) {
        StringBuilder codeBuilder = new StringBuilder();
        
        // Name
        codeBuilder.append("Name: ").append(name).append("\n");
        
        // Favorite programming language
        codeBuilder.append("Favorite programming language: ").append(language).append("\n");
        
        // Favorite operating system
        codeBuilder.append("Favorite operating system: ").append(os).append("\n");
        
        // Geekiness level
        codeBuilder.append("Geekiness level: ").append(geekiness).append("\n");
        
        return codeBuilder.toString();
    }
}

File: code_challenge_solution/c#/74 Geekcode generator.cs
using System;

namespace GeekcodeGenerator
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Geekcode Generator");
            Console.WriteLine("==================");

            Console.Write("Enter your name: ");
            string name = Console.ReadLine();

            Console.Write("Enter your age: ");
            int age = Int32.Parse(Console.ReadLine());

            Console.Write("Enter your favorite programming language: ");
            string favLanguage = Console.ReadLine();

            Console.Write("Enter your favorite game: ");
            string favGame = Console.ReadLine();

            Console.Write("Enter the number of hours you spend coding daily: ");
            int hoursCoding = Int32.Parse(Console.ReadLine());

            Console.WriteLine();

            Console.WriteLine("Your Geekcode Signature:");
            Console.WriteLine("------------------------");

            string geekcode = "GEEK-" + name.Substring(0, 3).ToUpper() + "-" + age.ToString() + "-" + favLanguage.Substring(0, 3).ToUpper() + "-" + favGame.Substring(0, 3).ToUpper() + "-" + hoursCoding.ToString();

            Console.WriteLine(geekcode);
        }
    }
}